### PR TITLE
Subset sql query builders and modifications to generate cohorts

### PR DESCRIPTION
**Pull request by azimov**
This implements demographics and cohort subsetting operations including `addCohortSubsetDefinition` and modifications for cohort generation in manner that treats base cohorts and subsets as similar entities.

This is a fully working implementation however, what is now needed:

*  More eyes to spot obvious mistakes
*  Checking that the logic of all the Subset operations is correct by validating cohorts
*  More examples of how to build complex queries/definitions

### CODE REVIEW

File: tests/testthat/test-Subsets.R [hunk 1]
```diff
@@ -60,22 +66,95 @@ test_that("Subset definition", {
...
+                            cohortIds = 11,
+                            cohortCombinationOperator = "all",
+                            negate = FALSE,
+                            startWindow = createSubsetCohortWindow(-999999, 99999, "cohortStart"),
+                            end = createSubsetCohortWindow(-999999, 99999, "cohortEnd"))
   expect_false(testDemoSubset2$isEqualTo(testDemoSubset))
+})
+
+
+test_that("subset generation", {
```
Comment (azimov) [inline tests/testthat/test-Subsets.R:96]: See here for example of full cohort subset generation

**Review by schuemie — APPROVED** [review]

**Comment by anthonysena** [comment]
Thanks @schuemie for the approval - @chrisknoll and I are still doing a review/test of this functionality so we'll hold on merging this until that work is complete.

**Comment by azimov** [comment]
@schuemie i didn't mean to request a re-review, i just have fat fingers

**Comment by azimov** [comment]
@anthonysena This PR should be ready to merge into develop - the vignette has now been updated to cover most functionality

**Pull request merged**